packages/finance/src/price/mod.rs

```diff
@@ -66,23 +66,45 @@ where
 {
     #[track_caller]
     fn new(amount: Coin<C>, amount_quote: Coin<QuoteC>) -> Self {
-        let (amount_normalized, amount_quote_normalized) = amount.into_coprime_with(amount_quote);
-        let res = Self {
+        let res: Self = Self::new_inner(amount, amount_quote);
+
+        #[cfg(debug_assertions)]
+        res.invariant_held().unwrap();
+
+        res
+    }
+
+    #[cfg(feature = "testing")]
+    pub const fn unchecked(amount: Coin<C>, amount_quote: Coin<QuoteC>) -> Self {
+        Self::new_inner(amount, amount_quote)
+    }
+
+    const fn new_inner(amount: Coin<C>, amount_quote: Coin<QuoteC>) -> Self {
+        let (amount_normalized, amount_quote_normalized): (Coin<C>, Coin<QuoteC>) =
+            amount.into_coprime_with(amount_quote);
+
+        Self {
             amount: amount_normalized,
             amount_quote: amount_quote_normalized,
-        };
-        debug_assert_eq!(Ok(()), res.invariant_held());
-        res
+        }
     }
 
     /// Returns a new [`Price`] which represents identity mapped, one to one, currency pair.
-    pub fn identity() -> Self {
+    pub const fn identity() -> Self {
         Self {
             amount: Coin::new(1),
             amount_quote: Coin::new(1),
         }
     }
 
+    pub const fn base_amount(&self) -> Coin<C> {
+        self.amount
+    }
+
+    pub const fn quote_amount(&self) -> Coin<QuoteC> {
+        self.amount_quote
+    }
+
```

Comment:
Price is not a bag of two amounts everyone should know about. please remove. Moreover you use them only for tests

Reply:
That would reintroduce more constants in the tests again, but I'll do as you wish.